docs(custom-elements): add externalRuntime config option (#924)
ionic-team/stencil-site: src/docs/output-targets/custom-elements.md

> Note: At this time, components created not using JSX may not be automatically
  defined. This is a known limitation of the API and users should be aware of
  it

> ### externalRuntime

Reviewer: Taking a step back, I took a look at the diff of using `externalRuntime` in a default Stencil project (`npm init stencil component`) this morning where I built the project twice with the following Stencil config output target settings:
```
    {
      type: 'dist-custom-elements',
      externalRuntime: false,
    },
    {
      type: 'dist-custom-elements-bundle',
      externalRuntime: false,
    },
```
```
    {
      type: 'dist-custom-elements',
      externalRuntime: true,
    },
    {
      type: 'dist-custom-elements-bundle',
      externalRuntime: true,
    },
```

It seems like this flag really only does anything for CEB. Is that right? Should we re-evaluate if this flag should exist for the CE output target?

Author: @rwaskiewicz I went and double-checked this and was seeing differences in `dist-custom-elements` with the flag on and off. Most notably is the bundling where, when `true`, the Stencil core code is not included in the bundle

Author: @rwaskiewicz I didn't run a diff, I was just looking at the output code when running with `externalRuntime: true` and `externalRuntime: false`. Were you not talking about the output? Maybe we can sync on this tomorrow morning?

When `true`, this flag results in the following behaviors:

1. Minification will follow what is specified in the [Stencil config](docs/config#minifyJs).
2. Filenames will not be hashed

> 3. When resolving all `@stencil/core/*` imports as a part of the rollup process all of these resolved modules will be marked as external and therefore not included in the bundle.

Author: Open to input here on more context or a way to group all these behaviors into a good description for what this really does 🤷‍♂️ Behaviors just feel disconnected from each other

Reviewer 2: Possibly it would be good to rephrase this along the same lines as the other entries in this list:

```suggestion
When `true`, this flag results in the following behaviors:

1. Minification will follow what is specified in the [Stencil config](docs/config#minifyJs).
2. Filenames will not be hashed
3. All imports from packages under `@stencil/core/*` will be marked as external and therefore not included in the bundle by Rollup.
```
